### ISSUE TITLE
Companion window --> repositioning button / drag handle

### ISSUE THREAD

**Issue by camillevilla** [ProjectMirador/mirador#1754]
Probably depends on pattern set up in #1752 (initial popping out implementation) 

![companion window on right](https://user-images.githubusercontent.com/5402927/52004705-a12aeb80-247c-11e9-80a9-8f8eb93df296.png)

![companion window bottom](https://user-images.githubusercontent.com/5402927/52005808-77bf8f00-247f-11e9-8a55-b0097f9b8de7.png)

### Resources
- [Interactive design](https://xd.adobe.com/view/1a628598-6843-4d25-67cb-d9e42eff6617-6c53/screen/4442ab28-e7a0-4ad4-962d-74506597c24d/Companion-right-default)
- Overall window layout diagram in #1715

### Acceptance criteria (still finalizing with design team) 
- [ ] Clicking and dragging the companion window to the bottom
  - [ ] Closes the right companion window
  - [ ] Moves content to the bottom companion window
- [ ] The image viewing area remains the same size (e.g. image is not clipped); total window area increases to accommodate the companion window

### Out of scope
- Visual niceties for dragging the window over (e.g. ghost boxes) 
![ghost-box](https://user-images.githubusercontent.com/5402927/52005700-26170480-247f-11e9-89ee-bc68780d5c7a.png)
- Keyboard controls (waiting for larger design / structure)

**Comment by camillevilla** [comment]
@ggeisler @jvine I just want to clarify intended behavior for dragging the window to the right area because it isn't in the interactive mockups. If dragging left:
- the sidebar pops open the contents on the left
- window state should look like this again, no explicit "companion window" label

![screen shot 2019-01-30 at 10 13 05 am](https://user-images.githubusercontent.com/5402927/52007591-9c1d6a80-2483-11e9-867e-e4cbd5b5ebb7.png)

**Comment by jvine** [comment]
I don't believe there is a drag-left for the companion window. Windows can be moved to the right or bottom of the primary window. Primary window consists of image + information panels. Companion window doesn't pop back into the info panels area. Logically if companion windows were to be dragged left, then the result would be positioned outside the primary window, to the left of the sidebar menu, taking the full height of the current layout.

**Comment by camillevilla** [comment]
Thanks for the confirmation. We will not be implementing or designing drag-left. 🎉

**Comment by charbugs** [comment]
@jvine @ggeisler 

Can the bottom companion window and the thumbnail navigation be open at the same time? If not, what should happen if the companion window opens while the thumbnail navigation is already open (and vice versa)?

**Comment by jvine** [comment]
@charbugs Yes, they can both be open at the same time. The thumbnail nav is essentially just another companion window, and would be stacked below (or beside) the canvas with any other panels. Here are a couple of screenshots showing stacked layouts.

<img width="960" alt="screen shot 2019-02-22 at 8 02 20 am" src="https://user-images.githubusercontent.com/6945691/53254533-629de080-3678-11e9-91c7-cdfe383d7d26.png">

<img width="963" alt="screen shot 2019-02-22 at 8 04 26 am" src="https://user-images.githubusercontent.com/6945691/53254603-86612680-3678-11e9-832d-f8409a7fd8f4.png">

**Comment by charbugs** [comment]
And the right companions expand to the bottom while the bottom companions shrinking, right?

**Comment by jvine** [comment]
See also #1760 - the side panels maintain the combined height of canvas + bottom panels, and the bottom panel maintains the width of the canvas alone.

**Comment by charbugs** [comment]
Ok, I see. #1760 is revealing. Thank you!

**Comment by jvine** [comment]
@charbugs We are working on this right now and wondering if we need to make a change. We have been thinking of the thumbnail strip as just another companion window, but looking at it in more detail, it seems like it has special characteristics that make the above pattern unsatisfactory. We are wondering if it's possible to pin the thumbnail strip at the bottom of the window, and make it full width. Something roughly like this (ignoring the shadow):

<img width="1157" alt="screen shot 2019-02-22 at 10 37 22 am" src="https://user-images.githubusercontent.com/6945691/53263324-df878500-368d-11e9-8c02-b1a32e75ffa5.png">

**Comment by charbugs** [comment]
Yes, I think it's possible. Thumbnail strips will then have an extra area on window bottom.

**Comment by camillevilla** [comment]
Cleared out assignees until we can re-prioritize this issue.

**Comment by marlo-longley** [comment]
Audit:
I am not sure if drag was ever implemented. Currently it doesn't exist.
There is no drag and drop for the companion windows, only the move button, which pops the window into different positions.
I personally like the icons in this ticket better than the ones currently in Mirador. 4 I think however that the popping button is working fine and is a simpler interaction, and nobody has asked for drag and drop over this period.  
Thanks for the work on this ticket in the past.
